ethportal-api/src/types/cli.rs

```diff
@@ -78,7 +78,7 @@ pub struct TrinConfig {
     pub web3_ipc_path: PathBuf,
 
     #[arg(
-        default_value = DEFAULT_DISCOVERY_PORT,
+        default_value_t = DEFAULT_DISCOVERY_PORT,
```

Comment:
This looks like a typo? I don't see `default_value_t` listed under the available [magic methods](https://docs.rs/structopt/latest/structopt/#magical-methods)

Reply:
I'm confused. Why did you reference `structopt` crate? I think we are using [`clap`](https://docs.rs/clap/latest/clap/). Am I missing something? (sorry, I started learning rust about a month ago)

Here is the documentation about attributes: https://docs.rs/clap/latest/clap/_derive/index.html#arg-attributes, but it might not be very clear what is the difference.

I found this [explanation](https://github.com/clap-rs/clap/discussions/3640) and in short, the `default_value` works on pre-parsed values (strings), while `default_value_t` works on post-parsed values (actual data type)

Reply:
Ahh, sorry my bad, I had it backwards. Yup, you're right we're using `clap` not `structopt` now.